_drafts/listas-desde-cero.md

```diff
@@ -0,0 +1,359 @@
+---
+layout: post
+title: Listas desde Cero
+subtitle: Una poco común introducción a la programación funcional
+---
+
+Imaginemos un ruby extraño:
+
+- Sin estructuras de datos como arrays o hashes.
+- Sin loops nativos como `for`, `while` o `loop`.
+- Sin clases: sólo funciones.
+
+La idea de este post es una prueba de concepto. La idea es explicar algunos conceptos de la programación funcional que no se ven tanto en la vida diaria pero que ayudan mucho a pensar en ese sentido. Imaginémonos en una situación tan limitada que debemos reconstruir — desde 0 — las cosas que siempre damos por sentadas en ruby únicamente con funciones.
+
+## Iteraciones
+
+Comencemos por una simple iteración. Tenemos que conseguir la suma de números en el rango de `x` a `y`. Siempre hemos usado bucles (`while`, `loop`) o métodos del módulo enumerable (`each`, `map`, `reduce`) para ejecutar algo repetidas veces. Por ejemplo, si tuviésemos que implementar esto:
+
+```ruby
+def sum_numbers(x, y)
+end
+
+sum_numbers(1, 10) # Expected: 55
+```
+
+Usaríamos posiblemente un `each` que itera sobre un rango acumulando una suma en una variable o algo similar. ¿Cómo lo haríamos sin estas herramientas que siempre usamos? (En serio, piensa un momento sobre esto sin continuar)
+
+Si no has hallado ya la respuesta — o sí pero quieres seguir leyendo —, pensemos en esa función de factorial que alguna vez hicimos — mientras aprendíamos o de otro modo — para demostrar cómo una función puede llamarse a sí misma:
```

Comment:
Oh, me refería a ésta: https://gist.github.com/fmeyer/289467#file-evolution-of-a-python-programmer-py-L1. En realidad me refería a la típica implementación recursiva del factorial que enseñan en algunos sitios.